easybuild/easyblocks/x/xalt.py

```diff
@@ -0,0 +1,160 @@
+            raise EasyBuildError(error_msg)
+
+        # GPU tracking
+        if self.cfg['gpu_tracking'] is True:
+            # User enabled
+            self.cfg.update('configopts', '--with-trackGPU=yes')
+        elif self.cfg['gpu_tracking'] is None:
+            # Default value, enable GPU tracking if nvml.h is present
+            # and the CUDA module is loaded
+            cuda_root = get_software_root('CUDA')
+            if not cuda_root:
+                # Try using the system CUDA
+                cuda_root = '/usr/local/cuda'
+            nvml_h = os.path.join(cuda_root, "include", "nvml.h")
+            if os.path.isfile(nvml_h):
+                self.cfg.update('configopts', '--with-trackGPU=yes')
+                self.cfg['gpu_tracking'] = True
+        else:
+            # User disabled
+            self.cfg.update('configopts', '--with-trackGPU=no')
+
+        # MySQL
+        if self.cfg['mysql'] is True:
+            self.cfg.update('configopts', '--with-MySQL=yes')
+        else:
+            self.cfg.update('configopts', '--with-MySQL=no')
+
+        # Configure
+        super(EB_XALT, self).configure_step()
+
+    def make_module_extra(self, *args, **kwargs):
+        txt = super(EB_XALT, self).make_module_extra(*args, **kwargs)
+
+        txt += self.module_generator.prepend_paths('LD_PRELOAD', 'lib64/libxalt_init.so')
+        txt += self.module_generator.prepend_paths('SINGULARITY_BINDPATH', '')
+        txt += self.module_generator.prepend_paths('SINGULARITYENV_LD_PRELOAD', 'lib64/libxalt_init.so')
+        txt += self.module_generator.set_environment('XALT_DIR', self.installdir)
+        txt += self.module_generator.set_environment('XALT_ETC_DIR', '%s' % os.path.join(self.installdir, 'etc'))
+        txt += self.module_generator.set_environment('XALT_EXECUTABLE_TRACKING',
+                                                     ('no', 'yes')[bool(self.cfg['executable_tracking'])])
+        txt += self.module_generator.set_environment('XALT_GPU_TRACKING',
+                                                     ('no', 'yes')[bool(self.cfg['gpu_tracking'])])
+        if self.cfg['transmission'].lower() == 'curl' and self.cfg['logging_url']:
+            txt += self.module_generator.set_environment('XALT_LOGGING_URL', self.cfg['logging_url'])
+        txt += self.module_generator.set_environment('XALT_SCALAR_SAMPLING',
+                                                     ('no', 'yes')[bool(self.cfg['scalar_sampling'])])
+
+        return txt
+
+    def make_module_req_guess(self):
+        """ Limit default guess paths
+        Do not set CPATH, LD_LIBRARY_PATH, LIBRARY_PATH, include sbin in PATH """
+        return {'COMPILER_PATH': 'bin',
+                'PATH': 'bin'}
+
+    def sanity_check_step(self):
+        """Custom sanity check"""
+        custom_paths = {
+            'files': ['lib64/libxalt_init.%s' % get_shared_lib_ext()],
+            'dirs': ['lib64'],
```

Comment:
Checking for a non-empty `lib64` directory isn't very useful if we're also checking for a specific file in this directory.

There's a couple of other important files we *should* be checking for here though: `bin/ld`, `bin/ld.gold`, and also the `libexec` and `sbin` directories.